### PR TITLE
refactor(pull): wip trying to replace pull with pull-v2

### PR DESCRIPTION
**Pull request by adrienaury**
THIS IS A WORK IN PROGRESS

Complete refactoring of the pull engine, but most of venom tests fail!

### CODE REVIEW

**Comment by adrienaury** [comment]
Remaining tests in error :

```
 • pull multi relation (tests/workspace/pull/multi_relation.yml)
        • pull-exhaust-multi-relations FAILURE
 • pull with file filters (tests/workspace/pull/pull_with_file_filters.yml)
        • pull-values-with-reversed-order FAILURE
        • pull-benchmark-two-tables FAILURE
 • pull statistics (tests/workspace/pull/stats.yml)
        • pull-with-filter-from-file-and-where-clause FAILURE
 • pull with where SQL clause (tests/workspace/pull/with_where_clause.yml)
        • pull-with-empty-where-clause FAILURE
```

**Comment by adrienaury** [comment]
Remaining tests in error :

```
 • pull with file filters (tests/workspace/pull/pull_with_file_filters.yml)
        • pull-values-with-reversed-order FAILURE
 • pull statistics (tests/workspace/pull/stats.yml)
        • pull-with-filter-from-file-and-where-clause FAILURE
 • pull with where SQL clause (tests/workspace/pull/with_where_clause.yml)
        • pull-with-empty-where-clause FAILURE
```

**Comment by adrienaury** [comment]
Only one test remains in failure

```
 • pull statistics (tests/workspace/pull/stats.yml)
        • pull-with-filter-from-file-and-where-clause FAILURE
```

File: pkg/pull/driver_test.go [hunk 1]
```diff
@@ -68,8 +66,7 @@ func LoadTest(filename string) (*Test, error) {
 
 func RunTest(t *testing.T, test *Test) {
 	t.Helper()
-	over.New(zerolog.New(os.Stderr))
```
Comment (Baguettte) [inline pkg/pull/driver_test.go]: Incompatible with stats

**Comment by Baguettte** [comment]
Nore more tests in failure.

**Pull request merged**